### PR TITLE
add check for valid slug fixes #2946

### LINKED ISSUE #2946: Rapid Experiment with no/invalid slug should return a validation error
If we enter something for name that will slugify to an empty slug, we have to throw a validation error.

Example:

name = `","`
slug = `""`

### PR DESCRIPTION
**Pull request by tiftran**

### CODE REVIEW

File: app/experimenter/experiments/api/v3/serializers.py [hunk 1]
```diff
@@ -30,6 +30,14 @@ class Meta:
             "audience",
         )
 
+    def validate_name(self, name):
+        slug = slugify(name)
+        if slug:
+            return name
+        raise serializers.ValidationError(
+            ["Name needs to contains alphanumeric characters"]
```
Comment (pdehaan) [inline app/experimenter/experiments/api/v3/serializers.py:38]: nit: Should this be "contain" singular? (here and then in the assert below)

File: app/experimenter/experiments/tests/api/v3/test_serializers.py [hunk 2]
```diff
@@ -156,3 +156,21 @@ def test_serializer_creates_changelog_for_updates(self):
                 changed_values=changed_values,
             ).exists()
         )
+
+    def test_serializer_returns_errors_for_non_alpha_numeric_name(self):
+
+        data = {
+            "name": "!!!!!!!!!!!!!!!",
```
Comment (pdehaan) [inline app/experimenter/experiments/tests/api/v3/test_serializers.py:163]: I should probably just verify on Experimenter Stage using un-rapid experiments, but... Do we have somewhere that verifies that a slug isn't already used? Or do name collisions on slugs not matter?

Comment (tiftran) [reply app/experimenter/experiments/tests/api/v3/test_serializers.py:163]: oooh you're right... should check for slug uniqueness as well

File: app/experimenter/experiments/api/v3/serializers.py (hunk 1 above)
Comment (tiftran) [reply app/experimenter/experiments/api/v3/serializers.py:38]: gah, bad grammar 😅

File: app/experimenter/experiments/tests/api/v3/test_serializers.py (hunk 2 above)
Comment (jaredlockhart) [reply app/experimenter/experiments/tests/api/v3/test_serializers.py:163]: Good catch @pdehaan let's do that as a followup: https://github.com/mozilla/experimenter/issues/2959

**Pull request merged**